### PR TITLE
Handle large write batches

### PR DESCRIPTION
**Pull request by wilhuff**
This addresses #208.

While this works, I'm not yet sure this approach performs well enough to merge as-is. For example, we could make the two methods I changed to use direct load to use the inline method that others use.

I've sent this out to get feedback on this before I invest time in benchmarking.

### CODE REVIEW

**Comment by wilhuff** [comment]
While testing the approach with `SQLiteStatement.executeForBlobFileDescriptor()` it turned out that this approach wasn't reliable. For reasons I didn't get to the bottom of, this would occasionally return `null` even for rows that existed and had a value.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteMutationQueue.java [hunk 1]
```diff
@@ -399,12 +406,76 @@ public void performConsistencyCheck() {
...
+  private MutationBatch decodeInlineMutationBatch(int batchId, byte[] bytes) {
     try {
+      if (bytes.length < BLOB_MAX_INLINE_LENGTH) {
+        return serializer.decodeMutationBatch(
+            com.google.firebase.firestore.proto.WriteBatch.parseFrom(bytes));
+      }
+
+      BlobAccumulator accumulator = new BlobAccumulator(bytes);
+      while (accumulator.more) {
+        int start = accumulator.numChunks() * BLOB_MAX_INLINE_LENGTH + 1;
```
[the remark this answers is not in the document]
Comment (wilhuff) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteMutationQueue.java]: Done. Added notes about how SUBSTR is 1-based as well.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteMutationQueue.java [hunk 2]
```diff
@@ -399,12 +406,76 @@ public void performConsistencyCheck() {
...
+
+        db.query("SELECT SUBSTR(mutations, ?, ?) FROM mutations WHERE uid = ? AND batch_id = ?")
```
[the remark this answers is not in the document]
Comment (wilhuff) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteMutationQueue.java]: I don't think this works. The 2 MB limit does not apply per column but per row. For the purposes of staying under the cursor row limit `SELECT SUBSTR(foo, 1, 1000), SUBSTR(foo, 1001, 1000)` is not functionally different from `SELECT SUBSTR(foo, 1, 2000)`.

[the remark this answers is not in the document]
Comment (var-const) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteMutationQueue.java]: Ah, I see, thanks for explaining.

**Pull request merged**